### PR TITLE
Bring RTMDetInst Optimization and tile unit test to Dev

### PR DESCRIPTION
**Pull request by eugene123tw**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/model/detection.py [hunk 1]
```diff
@@ -443,7 +443,7 @@ def _customize_outputs(
                 tv_tensors.BoundingBoxes(
                     output.pred_instances.bboxes,
                     format="XYXY",
-                    canvas_size=output.img_shape,
+                    canvas_size=output.ori_shape,
```
Comment (vinnamkim) [inline src/otx/core/model/detection.py:446]: Why is this changed? `img_shape` is the final image shape which lies in front of model's `forward()`. On the other hand `ori_shape` is the image shape where we can get by decoding it from the file.

**Review by jaegukhyun — APPROVED** [review]

File: src/otx/core/model/detection.py (hunk 1 above)
Comment (eugene123tw) [reply src/otx/core/model/detection.py:446]: `img_shape` is the input image size and not the original image size. Predicted boxes are rescaled to the original image size, so they are not within the canvas size.

Comment (vinnamkim) [reply src/otx/core/model/detection.py:446]: Do you mean that `output.pred_instances.bboxes` is rescaled to the original input image size?

Comment (eugene123tw) [reply src/otx/core/model/detection.py:446]: Yes, `output.pred_instances.bboxes` is in the same scaled as the original image size `ori_shape`. Rescaling happened in `mmdet.predict`

**Review by vinnamkim — APPROVED** [review]
